### PR TITLE
gadget,o/devicestate: set implicit values for schema and role directly instead of relying on Effective* accessors

### PR DESCRIPTION
**Pull request by pedronis**
Notice that the implicit role system-boot for label system-boot makes sense only for UC16/18 devices.

### CODE REVIEW

**Comment by pedronis** [comment]
based on #9748

File: gadget/gadget.go [hunk 1]
```diff
@@ -486,14 +508,25 @@ func validateVolume(name string, vol *Volume, model Model) error {
 	return validateCrossVolumeStructure(structures, knownStructures)
 }
 
+// isMBR returns whether the structure is the MBR and  can be used before setImplicitForVolume
```
Comment (bboozzoo) [inline gadget/gadget.go:511]: s/and  can/and can/

**Review by bboozzoo — APPROVED** [review]
LGTM

File: gadget/update.go [hunk 2]
```diff
@@ -136,6 +136,10 @@ func Update(old, new GadgetData, rollbackDirPath string, updatePolicy UpdatePoli
 		return err
 	}
 
+	if oldVol.Schema == "" || newVol.Schema == "" {
+		panic(fmt.Sprintf("unset volume schemas: old: %q new: %q", oldVol.Schema, newVol.Schema))
```
Comment (mvo5) [inline gadget/update.go:140]: Same question as Maciej above - should this be an internal error ?

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**